tests/Examples.hs

```diff
@@ -0,0 +1,1053 @@
+
+_List_reverse_1 :: TestTree
+_List_reverse_1 = Test.Tasty.HUnit.testCase "Example #1" (do
+    e <- Util.code [NeatInterpolation.text|
+./Prelude/List/reverse Integer ([] : List Integer)
+|]
+    Test.Tasty.HUnit.assertEqual "" (Util.normalize' e) "[] : List Integer" )
+
+_List_shifted_0 :: TestTree
+_List_shifted_0 = Test.Tasty.HUnit.testCase "Example #0" (do
+    e <- Util.code [NeatInterpolation.text|
+./Prelude/List/shifted
+Bool
+(   [   [   { index = +0, value = True  }
+        ,   { index = +1, value = True  }
+        ,   { index = +2, value = True  }
+        ]   : List { index : Natural, value : Bool }
+    ,   [   { index = +0, value = False }
+        ,   { index = +1, value = False }
+        ]   : List { index : Natural, value : Bool }
+    ,   [   { index = +0, value = True  }
+        ,   { index = +1, value = True  }
+        ,   { index = +2, value = True  }
+        ,   { index = +3, value = True  }
+        ]   : List { index : Natural, value : Bool }
+    ]   : List (List { index : Natural, value : Bool })
+)
+|]
+    Test.Tasty.HUnit.assertEqual "" (Util.normalize' e) "[{ index = +0, value = True }, { index = +1, value = True }, { index = +2, value = True }, { index = +3, value = False }, { index = +4, value = False }, { index = +5, value = True }, { index = +6, value = True }, { index = +7, value = True }, { index = +8, value = True }] : List { index : Natural, value : Bool }" )
+
+_List_shifted_1 :: TestTree
+_List_shifted_1 = Test.Tasty.HUnit.testCase "Example #1" (do
+    e <- Util.code [NeatInterpolation.text|
+./Prelude/List/shifted Bool ([] : List (List { index : Natural, value : Bool }))
+|]
+    Test.Tasty.HUnit.assertEqual "" (Util.normalize' e) "[] : List { index : Natural, value : Bool }" )
+
+_List_unzip_0 :: TestTree
+_List_unzip_0 = Test.Tasty.HUnit.testCase "Example #0" (do
+    e <- Util.code [NeatInterpolation.text|
+./Prelude/List/unzip
+Text
+Bool
+(   [   { _1 = "ABC", _2 = True  }
+    ,   { _1 = "DEF", _2 = False }
+    ,   { _1 = "GHI", _2 = True  }
+    ]   : List { _1 : Text, _2 : Bool }
+)
+|]
+    Test.Tasty.HUnit.assertEqual "" (Util.normalize' e) "{ _1 = [\"ABC\", \"DEF\", \"GHI\"] : List Text, _2 = [True, False, True] : List Bool }" )
+
+_List_unzip_1 :: TestTree
+_List_unzip_1 = Test.Tasty.HUnit.testCase "Example #1" (do
+    e <- Util.code [NeatInterpolation.text|
+./Prelude/List/unzip Text Bool ([] : List { _1 : Text, _2 : Bool })
+|]
+    Test.Tasty.HUnit.assertEqual "" (Util.normalize' e) "{ _1 = [] : List Text, _2 = [] : List Bool }" )
+
+_Monoid_00 :: TestTree
+_Monoid_00 = Test.Tasty.HUnit.testCase "Example #0" (do
+    _ <- Util.code [NeatInterpolation.text|
+./Prelude/Bool/and
+    : ./Prelude/Monoid Bool
+|]
+    return () )
+
+_Monoid_01 :: TestTree
+_Monoid_01 = Test.Tasty.HUnit.testCase "Example #1" (do
+    _ <- Util.code [NeatInterpolation.text|
+./Prelude/Bool/or
+    : ./Prelude/Monoid Bool
+|]
+    return () )
```

Comment:
We could avoid the `do` completely, using `(void $ Util.code [NeatInterpolation.text| ... |])` or we could have a new assertion, e.g. `assertTypeChecks [|NeatInterpolation.text| ... |]`.  That would make it more obvious what's the intention